server/groundstation.py

```diff
@@ -82,10 +85,36 @@ def uav_thread(self):
             time.sleep(0.1)
 
     def image_thread(self):
-        while True:
-            run = self.image.retreive_images()
-            # print("Image:", run)
-            time.sleep(0.1)
+        if self.config["socket"]["connection"]:  # Initialize a socket connection
+            with socket.socket(socket.AF_INET, socket.SOCK_STREAM) as s:
+                s.bind((self.config["socket"]["host"], self.config["socket"]["port"]))
+                s.listen()
+                conn, addr = s.accept()
+                with conn:
+                    print("Connection from:", addr)
+                    curr_image = ""
+                    while True:
+                        resp = conn.recv(1024)
+                        if not resp:
+                            break
+                        resp = resp.decode("utf-8")
+                        loc = resp.find(" ")
+                        if loc != -1:
+                            curr_image += resp[:loc]
+                            res = self.image.process_image(curr_image)
+                            curr_image = resp[loc + 1:]
+                            if res:
+                                print("Successfully Identified ODLC")
+                            conn.sendall(bytes("Image Complete", "utf-8"))
+                        else:
+                            curr_image += resp
```

Comment:
ODLC identification is one of the more important things to log, so you should probably add a log statement here.

Reply:
That's a good idea - Forgot about that when merging changes.